### PR TITLE
update dependency versions

### PR DESCRIPTION
**Pull request by albertshau**
Use non-snapshot versions of CDAP dependencies and update jackson
to a newer version.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -30,21 +30,21 @@
...
+    <hydrator.version>2.2.0</hydrator.version>
     <commons.version>3.8.1</commons.version>
     <salesforce.api.version>45.0.0</salesforce.api.version>
     <cometd.java.client.version>4.0.0</cometd.java.client.version>
     <antlr.version>4.7.2</antlr.version>
     <mockito.version>1.10.19</mockito.version>
     <commons.csv.version>1.6</commons.csv.version>
     <jackson.version>1.9.13</jackson.version>
-    <jackson2.version>2.6.7</jackson2.version>
+    <jackson2.version>2.9.9</jackson2.version>
```
Comment (CuriousVini) [inline pom.xml:47]: Why is this change needed?

File: pom.xml [hunk 2]
```diff
@@ -30,21 +30,21 @@
     <surefire.redirectTestOutputToFile>true</surefire.redirectTestOutputToFile>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <!-- version properties -->
-    <cdap.version>6.0.0-SNAPSHOT</cdap.version>
+    <cdap.version>6.0.0</cdap.version>
```
Comment (CuriousVini) [inline pom.xml:33]: shouldn't we rely on cdap 6.1.0-SNAPSHOT on develop similar to https://github.com/data-integrations/wrangler/blob/develop/pom.xml#L86?

File: pom.xml (hunk 1 above)
Comment (albertshau) [reply pom.xml]: 2.6.7 has a security bug

File: pom.xml (hunk 2 above)
Comment (albertshau) [reply pom.xml:33]: This plugin declares that it is compatible with 6.x versions of the data-pipeline app, which means this needs to be 6.0.0. If it were to use a new API in 6.1.0-SNAPSHOT and then were to be deployed on a 6.0.x version of CDAP, it would fail.

In general, I think we should change our process for plugins such that we don't update the dependent cdap version unless they need to use some new API, so that they're compatible with as many versions of CDAP as possible.

**Review by CuriousVini — APPROVED** [review]
